cms/src/builder.rs

```diff
@@ -167,28 +164,27 @@ where
     }
 }
 
-impl<'s, S> Builder for SignerInfoBuilder<'s, S>
-where
-    S: Keypair + DynSignatureAlgorithmIdentifier,
-{
-    type Signer = S;
+impl<'s> Builder for SignerInfoBuilder<'s> {
     type Output = SignerInfo;
 
-    fn signer(&self) -> &Self::Signer {
-        self.signer
-    }
-
     /// Calculate the data to be signed
     /// [RFC 5652 § 5.4](https://datatracker.ietf.org/doc/html/rfc5652#section-5.4)
     /// If an `external_message_digest` is passed in, it is assumed, that we are signing external
     /// content (see RFC 5652 § 5.2). In this case, the `eContent` in `EncapsulatedContentInfo`
     /// must be `None`.
-    fn finalize(&mut self) -> der::Result<Vec<u8>> {
+    fn finalize<S>(
+        &mut self,
+        _signer: &S,
+    ) -> core::result::Result<Vec<u8>, x509_cert::builder::Error>
```

Comment:
?

Reply:
That was just a rebase from this PR on a fresh master.
But I needed to change the signature of finalize because I attach the public key of the signer "late".
Getting the public from the keypair made it so that I could not get away with just a `der::Result`.

This is still a breaking change though.